### PR TITLE
[node-agent] Introduce `cmd/gardener-node-agent/app.go` with `manager.Manager` initialization

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
This PR introduces the `cmd/gardener-node-agent/app.go` with `manager.Manager` initialization.

If it will be started with a kubeconfig pointing to the well-known location of the bootstrap token file, it will use it to fetch the access token and write it to the disk. Afterwards, it will overwrite the kubeconfig to point to the access token file instead of the bootstrap token file. The rest config will be modified ad-hoc, hence, the bootstrap token will only be used for fetching the "real" access token.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/8023

**Special notes for your reviewer**:
/cc @oliver-goetz 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: cmd/gardener-node-agent/app/app.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	mgr, err := manager.New(restConfig, manager.Options{
+		Logger:                  log,
+		Scheme:                  kubernetes.SeedScheme,
+		GracefulShutdownTimeout: pointer.Duration(5 * time.Second),
+
+		HealthProbeBindAddress: net.JoinHostPort(cfg.Server.HealthProbes.BindAddress, strconv.Itoa(cfg.Server.HealthProbes.Port)),
+		Metrics: metricsserver.Options{
+			BindAddress:   net.JoinHostPort(cfg.Server.Metrics.BindAddress, strconv.Itoa(cfg.Server.Metrics.Port)),
+			ExtraHandlers: extraHandlers,
+		},
```
Comment (oliver-goetz) [inline cmd/gardener-node-agent/app/app.go:151]: There is neither a validation nor a defaulting for health probe and metrics port. Thus, there is a risk of creating node agent configs which make it fail on startup.

And just for confirmation. Node agent is opening a node port here. It is safe to assume that the node network is not publicly available, isn't it?

File: cmd/gardener-node-agent/app/app.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	token := secret.Data[resourcesv1alpha1.DataKeyToken]
+	if len(token) == 0 {
+		return fmt.Errorf("secret key %q does not exist or empty", resourcesv1alpha1.DataKeyToken)
+	}
+
+	log.Info("Successfully fetched access token from API server, writing it to disk", "path", nodeagentv1alpha1.TokenFilePath)
+	if err := os.WriteFile(nodeagentv1alpha1.TokenFilePath, token, 0600); err != nil {
+		return fmt.Errorf("unable to write access token to %s: %w", nodeagentv1alpha1.TokenFilePath, err)
+	}
+	log.Info("Token written to disk")
```
Comment (oliver-goetz) [inline cmd/gardener-node-agent/app/app.go]: We could move the token write part to the end of the function just before writing the kubeconfig. Then, we would not create an orphaned token on the file system if encoding of the kubeconfig fails.

**Review by oliver-goetz — COMMENTED** [review]
awesome 🚀 

We should not forget to add the node-agent image to `branch.cfg` and [image build prow job](https://github.com/gardener/ci-infra/blob/37e6655320944c697d9c7ce479bf2c5ab210824c/config/jobs/gardener/gardener-build-dev-images.yaml#L34) when we merge the PR.

**Comment by rfranzke** [comment]
> We should not forget to add the node-agent image to `branch.cfg` and [image build prow job](https://github.com/gardener/ci-infra/blob/37e6655320944c697d9c7ce479bf2c5ab210824c/config/jobs/gardener/gardener-build-dev-images.yaml#L34) when we merge the PR.

We can also wait a bit longer with this until we finally integrate the GNA into `gardenlet`. It won't be used before that.

File: cmd/gardener-node-agent/app/app.go (hunk 1 above)
Comment (rfranzke) [reply cmd/gardener-node-agent/app/app.go:151]: > There is neither a validation nor a defaulting for health probe and metrics port. Thus, there is a risk of creating node agent configs which make it fail on startup.

`gardenlet` will be the only user of GNA and its configuration - we probably won't deploy it with "broken" configuration. If end-users manually tamper with it - sure, they can break it, just like they can break anything else as well.

> And just for confirmation. Node agent is opening a node port here. It is safe to assume that the node network is not publicly available, isn't it?

Even if not, having the `kubelet` publicly available is probably the greater risk compared to the GNA health/metrics endpoint 😄

**Pull request merged**